tests/core/pyspec/eth2spec/test/altair/transition/test_transition.py

```diff
@@ -0,0 +1,85 @@
+from eth2spec.test.context import default_activation_threshold, default_balances
+from eth2spec.test.helpers.genesis import create_genesis_state
+from eth2spec.test.helpers.state import transition_unsigned_block
+from eth2spec.test.helpers.block import build_empty_block_for_next_slot
+from eth2spec.test.utils import vector_test
+from eth2spec.phase0 import spec as phase0
+from eth2spec.altair import spec
+
+
+class VersionedObject(spec.Container):
+    fork_version: spec.Version
+    obj: spec.Container
+
+
+def _with_tag(tag, obj):
+    return VersionedObject(fork_version=tag, obj=obj)
+
+
+def _to_hex(data):
+    return "0x" + bytes(data).hex()
+
+
+def transition_test(pre_state_spec=phase0, pre_version=spec.GENESIS_FORK_VERSION, post_version=spec.ALTAIR_FORK_VERSION, description=None, fork_epoch=None):
+    def _wrapper(test_fn):
+        @vector_test(description)
+        def _adapter(*args, **kwargs):
+            # test gen infra passes some unnecessary arguments
+            # clear here before proceeding...
+            kwargs.pop("phase", None)
+            kwargs.pop("bls_active", None)
+
+            def pre_tag(obj):
+                return _with_tag(pre_version, obj)
+
+            def post_tag(obj):
+                return _with_tag(post_version, obj)
+
+            # the `with_state` decorators pull in more phase
+            # machinery than we need so just re-create here...
+            state = create_genesis_state(
+                pre_state_spec, 
+                default_balances(pre_state_spec),
+                default_activation_threshold(pre_state_spec),
+            )
```

Comment:
could make another decorator that exposes the balances and activation threshold on the genesis state here... wasn't obvious how we can just use `with_state` decorator(s) here....

Reply:
ah, just saw this. It would be very valuable to be able to use `with_state`

Can we not use the standard decorator and if state is missing, use a default?

Reply:
so main issue was just that `with_state` assumes the wrapped function needs a `spec` and some `phases`:

https://github.com/ethereum/eth2.0-specs/blob/dev/tests/core/pyspec/eth2spec/test/context.py#L69-L87

we can either fork `with_state` and write the same thing w/o these parameters or define an adapter from the transition test to the format `with_state` expects....

i'll go look more closely to see if we can use `with_state` directly --- i think one thing for example is that we will have to specify a spec to use, which could likely just be the pre-fork spec and then `phases` could span the fork boundary; i'll go investigate a bit further